restarts when rebuild is hit

```warn: (watchly):(nStwXJV1n0jOM1Gw): deadline not set in config, defaulting to 30 minutes
github-info: started build: nStwXJV1n0jOM1Gw
github-info: build started
info: (watchly):(nStwXJV1n0jOM1Gw): provisioning
github-info: Asked to provide for git@github.com:gordallott/watchly.git
github-info: Asked to provide for git@github.com:watchlyhq/watchly.git
github-info: Got webhook event: status
github-warn: Could not handle event type: status
info: (watchly):(nStwXJV1n0jOM1Gw): running build: /tmp/ngbuild/builds/ngbuild-workspace-568250008/build.sh
info: (watchly):(nStwXJV1n0jOM1Gw): Command started, pid=2972
web-info: Starting monitoring of map[app:watchly token:nStwXJV1n0jOM1Gw]
crit: (watchly):(nStwXJV1n0jOM1Gw): Process exited but stdpipes are still open(zombied): 2972
info: (watchly):(nStwXJV1n0jOM1Gw): Build exited, waiting...
warn: (watchly):(nStwXJV1n0jOM1Gw): Build exited with non zero error code
warn: (watchly):(nStwXJV1n0jOM1Gw): Build exited with error: exit status 1
info: (watchly):(nStwXJV1n0jOM1Gw): this message stops a linter error
warn: (watchly):(SrlX0oMl3kEgLq0z): deadline not set in config, defaulting to 30 minutes
github-info: build started
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x469e17]
goroutine 20 [running]:
sync.(*RWMutex).RLock(0x0)
        /opt/go/src/sync/rwmutex.go:41 +0x27
github.com/watchly/ngbuild/core.(*BuildConfig).GetMetadata(0xc42010a680, 0x772b44, 0x10, 0x0, 0x0)
        /opt/go_workspace/src/github.com/watchly/ngbuild/core/buildconfig.go:22 +0x44
github.com/watchly/ngbuild/integrations/github.(*Github).updateBuildStatus(0xc42007f0e0, 0x8d8120, 0xc4203c41e0, 0x8d8500, 0xc420094a50)
        /opt/go_workspace/src/github.com/watchly/ngbuild/integrations/github/buildevents.go:14 +0xa6
github.com/watchly/ngbuild/integrations/github.(*Github).onBuildStarted(0xc42007f0e0, 0xc42024a360)
        /opt/go_workspace/src/github.com/watchly/ngbuild/integrations/github/buildevents.go:100 +0x320
github.com/watchly/ngbuild/integrations/github.(*Github).(github.com/watchly/ngbuild/integrations/github.onBuildStarted)-fm(0xc42024a360)
        /opt/go_workspace/src/github.com/watchly/ngbuild/integrations/github/github.go:196 +0x34
github.com/watchly/ngbuild/core.(*appbus).fireEvent(0xc4204cd800, 0xc420248b40, 0x36)
        /opt/go_workspace/src/github.com/watchly/ngbuild/core/appbus.go:133 +0x16d
github.com/watchly/ngbuild/core.(*appbus).coreloop(0xc4204cd800)
        /opt/go_workspace/src/github.com/watchly/ngbuild/core/appbus.go:112 +0xe5
created by github.com/watchly/ngbuild/core.newAppBus
        /opt/go_workspace/src/github.com/watchly/ngbuild/core/appbus.go:31 +0x11e
,.-~*´¨¯¨`*·~-.¸-(_NGBuild_)-,.-~*´¨¯¨`*·~-.¸
   Building your dreams, one step at a time
Visit the webUI on http://ngbuild.illuminaughty.io/web/status
info: Starting http listen server on :80
Available Integrations:
    Web
    github
    slack
Got authentication response
Found repositories:
Ayatiihq/Sentry
gordallott/agent 🔒🍴
gordallott/Cataclysm-DDA 🍴
gordallott/gdsstress
gordallott/glux
gordallott/glux-go
gordallott/godeps 🍴
gordallott/hue 🍴
gordallott/ini 🍴
gordallott/lint 🍴
gordallott/loggo 🍴
gordallott/ngbuild 🍴
gordallott/node-torrent 🍴
gordallott/testpostplzignore 🔒🍴
gordallott/testrepo
gordallott/twitch-announcer
gordallott/vimconfig
gordallott/watchly 🔒🍴
watchly/letsencrypt 🍴
watchly/material-ui 🍴
watchly/ngbuild
watchly/protobuf 🍴
watchly/redis-stateful-set 🍴```

--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,4 @@
 ngbuild
+ngbuild.json
+apps/
 integrations/slack/bin

--- a/core/buildconfig.go
+++ b/core/buildconfig.go
@@ -3,6 +3,7 @@ package core
 import (
 	"encoding/json"
 	"io/ioutil"
+	"sync"
 )
 
 // SetMetadata will set metadata
@@ -62,6 +63,7 @@ func UnmarshalBuildConfig(filename string) (*BuildConfig, error) {
 		return nil, err
 	}
 	conf := marshalledConf.Config
+	conf.m = &sync.RWMutex{}
 	conf.metadata = make(map[string]string)
 	// there isn't a nice way of copying a map in go.. so here we go
 	for key, value := range *marshalledConf.Metadata {

